src/file/format/dhall.rs

```diff
@@ -0,0 +1,55 @@
+use crate::map::Map;
+use std::error::Error;
+
+use crate::{
+    error::Unexpected,
+    value::{Value, ValueKind},
+    ConfigError,
+};
+
+pub fn parse(
+    uri: Option<&String>,
+    text: &str,
+) -> Result<Map<String, Value>, Box<dyn Error + Send + Sync>> {
+    let value = from_dhall_value(uri, serde_dhall::from_str(text).parse()?);
+    match value.kind {
+        ValueKind::Table(map) => Ok(map),
+        ValueKind::Nil => Err(Unexpected::Unit),
+        ValueKind::Boolean(value) => Err(Unexpected::Bool(value)),
+        ValueKind::Integer(value) => Err(Unexpected::Integer(value)),
+        ValueKind::Float(value) => Err(Unexpected::Float(value)),
+        ValueKind::String(value) => Err(Unexpected::Str(value)),
+        ValueKind::Array(value) => Err(Unexpected::Seq),
+    }
+    .map_err(|err| ConfigError::invalid_root(uri, err))
+    .map_err(|err| Box::new(err) as Box<dyn Error + Send + Sync>)
+}
+
+fn from_dhall_value(uri: Option<&String>, value: serde_dhall::SimpleValue) -> Value {
+    match value {
+        serde_dhall::SimpleValue::Num(num) => match num {
+            serde_dhall::NumKind::Bool(b) => Value::new(uri, ValueKind::Boolean(b)),
+            serde_dhall::NumKind::Natural(n) => Value::new(uri, ValueKind::Integer(n as i64)),
+            serde_dhall::NumKind::Integer(i) => Value::new(uri, ValueKind::Integer(i)),
```

Comment:
TOML seems to rely on inference while others are more explicitly typing to `ValueKind::I64(value)`, should I go with the latter?

https://github.com/mehcode/config-rs/blob/d7c165602199d045cdee3aab23ff69d7269de7e3/src/file/format/toml.rs#L24

https://github.com/mehcode/config-rs/blob/d7c165602199d045cdee3aab23ff69d7269de7e3/src/file/format/ron.rs#L34

https://github.com/mehcode/config-rs/blob/d7c165602199d045cdee3aab23ff69d7269de7e3/src/file/format/json.rs#L24-L26

https://github.com/mehcode/config-rs/blob/d7c165602199d045cdee3aab23ff69d7269de7e3/src/file/format/yaml.rs#L49